be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/Validator.scala

```diff
@@ -2,17 +2,46 @@ package ch.epfl.pop.pubsub.graph
 
 import akka.NotUsed
 import akka.stream.scaladsl.Flow
+import ch.epfl.pop.model.network.requests.lao.{JsonRpcRequestCreateLao, JsonRpcRequestStateLao, JsonRpcRequestUpdateLao}
+import ch.epfl.pop.model.network.requests.meeting.{JsonRpcRequestCreateMeeting, JsonRpcRequestStateMeeting}
+import ch.epfl.pop.model.network.requests.witness.JsonRpcRequestWitnessMessage
+import ch.epfl.pop.pubsub.graph.validators.LaoValidator.{validateCreateLao, validateStateLao, validateUpdateLao}
+import ch.epfl.pop.pubsub.graph.validators.MeetingValidator.{validateCreateMeeting, validateStateMeeting}
+import ch.epfl.pop.pubsub.graph.validators.RollCallValidator.{validateCloseRollCall, validateCreateRollCall, validateOpenRollCall, validateReopenRollCall}
+import ch.epfl.pop.pubsub.graph.validators.WitnessValidator.validateWitnessMessage
 
 object Validator {
 
   // FIXME implement schema
   def validateSchema(jsonString: JsonString): Either[JsonString, PipelineError] = Left(jsonString)
 
   // FIXME implement rpc validator
-  def validateJsonRpcContent(graphMessage: GraphMessage): GraphMessage = graphMessage
-
-  // FIXME implement message validator
-  def validateMessageContent(graphMessage: GraphMessage): GraphMessage = graphMessage
+  def validateJsonRpcContent(graphMessage: GraphMessage): GraphMessage = graphMessage match {
+    case Left(jsonRpcMessage) => jsonRpcMessage match {
+      case _ => ???
+    }
+    case graphMessage@_ => graphMessage
+  }
+
+  def validateMessageContent(graphMessage: GraphMessage): GraphMessage = graphMessage match {
+    case Left(jsonRpcMessage) => jsonRpcMessage match {
+      case message@(_: JsonRpcRequestCreateLao) => validateCreateLao(message)
+      case message@(_: JsonRpcRequestStateLao) => validateStateLao(message)
+      case message@(_: JsonRpcRequestUpdateLao) => validateUpdateLao(message)
+      case message@(_: JsonRpcRequestCreateMeeting) => validateCreateMeeting(message)
+      case message@(_: JsonRpcRequestStateMeeting) => validateStateMeeting(message)
+      case message@(_: JsonRpcRequestCreateLao) => validateCreateRollCall(message)
+      case message@(_: JsonRpcRequestStateLao) => validateOpenRollCall(message)
+      case message@(_: JsonRpcRequestUpdateLao) => validateReopenRollCall(message)
+      case message@(_: JsonRpcRequestUpdateLao) => validateCloseRollCall(message)
+      case message@(_: JsonRpcRequestWitnessMessage) => validateWitnessMessage(message)
+      case _ => Right(PipelineError(
+        ErrorCodes.SERVER_FAULT.id,
```

Comment:
I'm not sure this is strictly a server error. It might well be that whoever sent the message has a buggy front-end :) 
I feel this would be an INVALID_ACTION as far as the server is concerned.
Same remark applies to other similar cases below.

We had thought of the server error more as a way of reporting that we divided by 0 and crashed (for instance) than due to any data going across the network.

Reply:
Got it! :)